products/cloudbuild/api.yaml

```diff
@@ -503,3 +503,186 @@ objects:
                     have completed successfully. If `wait_for` is empty, this build step
                     will start when all previous build steps in the `Build.Steps` list
                     have completed successfully.
+          - !ruby/object:Api::Type::NestedObject
+            name: 'artifacts'
+            description: |
+              Artifacts produced by the build that should be uploaded upon successful completion of all build steps.
+            properties:
+              - !ruby/object:Api::Type::Array
+                name: 'images'
+                item_type: Api::Type::String
+                description: |
+                  A list of images to be pushed upon the successful completion of all build steps.
+
+                  The images will be pushed using the builder service account's credentials.
+
+                  The digests of the pushed images will be stored in the Build resource's results field.
+
+                  If any of the images fail to be pushed, the build is marked FAILURE.
+              - !ruby/object:Api::Type::NestedObject
+                name: 'objects'
+                description: |
+                  A list of objects to be uploaded to Cloud Storage upon successful completion of all build steps.
+
+                  Files in the workspace matching specified paths globs will be uploaded to the
+                  Cloud Storage location using the builder service account's credentials.
+
+                  The location and generation of the uploaded objects will be stored in the Build resource's results field.
+
+                  If any objects fail to be pushed, the build is marked FAILURE.
+                properties:
+                  - !ruby/object:Api::Type::String
+                    name: 'location'
+                    description: |
+                      Cloud Storage bucket and optional object path, in the form "gs://bucket/path/to/somewhere/".
+                      (see Bucket Name Requirements).
```

Comment:
I'd recommend either getting rid of this line or adding the place this links to. Right now standing alone it's not adding much value.